cmd/clusterctl/client/tree/util.go

```diff
@@ -102,7 +103,22 @@ func VirtualObject(namespace, kind, name string) *unstructured.Unstructured {
 				"annotations": map[string]interface{}{
 					VirtualObjectAnnotation: "True",
 				},
-				"uid": fmt.Sprintf("%s, %s/%s", gk, namespace, name),
+				"uid": fmt.Sprintf("%s, Kind=%s, %s/%s", gk, kind, namespace, name),
+			},
+		},
+	}
+}
+
+// ObjectReferenceObject return a new virtual object.
```

Comment:
nit (non-blocking)
```suggestion
// ObjectReferenceObject returns a new object referenced by the objectRef.
```

This function technically does not return a virtual object (the Group version and the annotations are not added for these objects).